api/fsm.go

```diff
@@ -159,7 +159,7 @@ func (x *Configuration) CheckValid() error {
         return MissingStatesConfigurationError
     }
     if x.StartingState == "" || !x.HasState(x.StartingState) {
```

Comment:
not only that, but actually, this needs splitting into two separate checks (done) so that the error message reflects the error of an empty state
thanks